quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/CachingPropertyMappers.java

```diff
@@ -56,17 +57,21 @@ public static PropertyMapper<?>[] getClusteringPropertyMappers() {
                     .build(),
               fromOption(CachingOptions.CACHE_EMBEDDED_MTLS_KEYSTORE.withRuntimeSpecificDefault(getDefaultKeystorePathValue()))
                     .paramLabel("file")
```

Comment:
```suggestion
                    .paramLabel("file")
                    .isEnabled(() -> isTrue(CACHE_EMBEDDED_MTLS_ENABLED), "property '%s' is enabled.".formatted(CACHE_EMBEDDED_MTLS_ENABLED.getKey()))
```

nitpick: You can leverage the `isEnabled()` functionality to have a unified approach to disabled options. The validator might only check for the presence of the password props.